auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,405 @@
+#  -*- encoding: utf-8 -*-
+##############################################################################
+#     OpenERP, Open Source Management Solution
+#     Copyright (C) 2004-2009 Tiny SPRL (<http://tiny.be>). All Rights Reserved
+#     Copyright 2015 Agile Business Group <http://www.agilebg.com>
+#
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import xmlrpclib
+import socket
+import os
+import time
+import datetime
+import base64
+import re
+try:
+    import pysftp
+except ImportError:
+    raise ImportError(
+        'This module needs pysftp to automaticly write backups to the FTP '
+        'through SFTP.Please install pysftp on your system.'
+        '(sudo pip install pysftp)'
+    )
+from openerp import models, fields, api, _
+from openerp.exceptions import except_orm, Warning
+from openerp import tools
+from openerp import netsvc
+import logging
+_logger = logging.getLogger(__name__)
+
+
+def execute(connector, method, *args):
+    res = False
+    try:
+        res = getattr(connector, method)(*args)
+    except socket.error as e:
+        raise e
+    return res
+
+
+class db_backup(models.Model):
+    _name = 'db.backup'
+
+    def get_connection(self, host, port):
+        uri = 'http://%s:%s' % (host, port)
+        return xmlrpclib.ServerProxy(uri + '/xmlrpc/db')
+
+    def get_db_list(self, host, port):
+        conn = self.get_connection(host, port)
+        db_list = execute(conn, 'list')
+        return db_list
+
+    @api.model
+    def _get_db_name(self):
+        return self.env.cr.dbname
+
+    # Columns local server
+    host = fields.Char(
+        string='Host', default='localhost', size=100, required=True)
+    port = fields.Char(
+        string='Port', default='8069', size=10, required=True)
```

Comment:
The port is available in the configuration of the server, and for host, localhost should always work, unless you explicitely disable listening on 127.0.0.1 (which is usually the only interface which is kept when securing a server behind a reverse proxy) -> I don't see why you need to configure this.